lib/src/cartesian/utils/transformInputData.ts

```diff
@@ -145,8 +146,13 @@ export const transformInputData = <
   });
 
   yKeys.forEach((yKey) => {
-    y[yKey].i = data.map((datum) => asNumber(datum[yKey]));
-    y[yKey].o = data.map((datum) => yScale(asNumber(datum[yKey])));
+    y[yKey].i = data.map((datum) => datum[yKey] as MaybeNumber);
+    y[yKey].o = data.map(
+      (datum) =>
+        (typeof datum[yKey] === "number"
```

Comment:
Do we want to also handle `NaN` since this would be `true` if you did something like `y.key: NaN`